RFS/src/main/java/com/rfs/common/LuceneDocumentsReader.java

```diff
@@ -35,7 +35,7 @@ public Flux<Document> readDocuments(Path luceneFilesBasePath, String indexName,
                 try {
                     reader.close();
                 } catch (IOException e) {
-                    logger.error("Failed to close IndexReader", e);
+                    throw new RuntimeException("Failed to close IndexReader", e);
```

Comment:
nit: you'll get snagged on a linter here.  If you use Lombok.sneakyThrows(e), you're fine.  It makes backtraces easier to understand and handle too.  Typed exception handling is a language construct, not a VM/runtime one - so you basically just say, "hey compiler, let me throw this thing & ignore your checks" - making this the same as a RuntimeException to callers.  This becomes especially critical once you have to unpack exceptions from async activities that could be wrapped numerous times otherwise (`completedException.cause().cause().instanceof(...) || completedException.cause().cause().cause().instanceof(...)` - remember that from the Divine Comedy ;) )

Reply:
Done!